### DISCUSSION TITLE
Trade entry price not as expected

### DISCUSSION THREAD

**Discussion by CadeHalcyon** [kernc/backtesting.py#766] [Q&A] [answered]
I am new to backtester.py and am trying out trade entries to see how they materialise, and currently they are not as expected.  I wrote the following code:

`
def next (self):

        Wf_Top = self.wf_Top  
        Wf_Low = self.wf_Low `
        
        if not self.position:
            if self.data.Close[-1] > Wf_Top[-1]: 
                self.sell()
        else:
            if self.data.Close[-1] < Wf_Low[-1]: 
                self.position.close()`

Note:  Wf_Top is the blue lambda line on the chart in the pic below:

<img width="1010" alt="Screenshot 2022-10-06 at 23 17 52" src="https://user-images.githubusercontent.com/115154598/194429945-d0ffd185-5a17-4d9a-860e-e397ca2f3b62.png">

My intention was for a sell trade to be entered above the blue line on the chart, however, the trade was entered well below the blue line, and also at a price point that doesn't even exist at the time the trade was entered (it is below any candlesticks).  Grateful for any help with this.  Thanks.

**Comment by kernc** [comment, marked as answer]
I suppose you were unable to check the FAQ.

It's got to do with [`Backtest(..., commission=...)`](https://kernc.github.io/backtesting.py/doc/backtesting/backtesting.html#backtesting.backtesting.Backtest). See e.g.: https://github.com/kernc/backtesting.py/issues/633 https://github.com/kernc/backtesting.py/issues/538

**Reply by CadeHalcyon** [reply]
Thanks
